internal/pkg/overlay/overlay.go

```diff
@@ -139,10 +140,18 @@ func OverlayInit(overlayName string) error {
 /*
 Build the given overlays for a node and create a Image for them
 */
-func BuildOverlay(nodeInfo node.NodeInfo, overlayNames []string) error {
+func BuildOverlay(nodeInfo node.NodeInfo, overlayNames []string, img_context ...string) error {
+	var context string
+	/* Check optional context argument. If missing, default to legacy. */
+	if len(img_context) == 0 {
+		context = "legacy"
+	} else {
+		context = img_context[0]
```

Comment:
if we are only using first item of img_context, shall we directly declare img_context as `string` ?
we just need to check whether img_context is set or not.

Reply:
My extremely limited understanding is that the optional arguments in a variadic function will always be in an array as there's no way to know how many things will be passed to it.